### PR TITLE
bisect--helper: `bisect_head` shell function in C

### PR DESCRIPTION
**Pull request by pranitbauva1997**
I have converted this function right now (its not really required) but I did it anyway because it was an easy one and I am getting a good feeling when everything's working fine!

### CODE REVIEW

File: builtin/bisect--helper.c [hunk 1]
```diff
@@ -78,6 +79,25 @@ static int write_terms(const char *bad, const char *good)
 	return (res < 0) ? -1 : 0;
 }
 
+char *bisect_head(void)
+{
+	FILE *fp;
+	static char buf[15];
+	struct strbuf head_name = STRBUF_INIT;
+	fp = fopen(git_path_bisect_head(), "r");
```
Comment (chriscool) [inline builtin/bisect--helper.c]: Are you sure it is necessary to open the file?

File: builtin/bisect--helper.c [hunk 2]
```diff
@@ -78,6 +79,25 @@ static int write_terms(const char *bad, const char *good)
...
+	if (fp) {
+		strbuf_addstr(&head_name, "BISECT_HEAD");
+		fclose(fp);
+	}
+	else
+		strbuf_addstr(&head_name, "HEAD");
+
+	strcpy(buf, head_name.buf);
```
Comment (chriscool) [inline builtin/bisect--helper.c]: It seems wasteful to use both a strbuf and a "static char buf[15]".
Couldn't the caller pass a strbuf and this function just add to this strbuf?

Comment (chriscool) [inline builtin/bisect--helper.c]: Or maybe this function could just return either "HEAD" or "BISECT_HEAD" and have the caller copy that if needed?

File: builtin/bisect--helper.c (hunk 1 above)
Comment (pranitbauva1997) [inline builtin/bisect--helper.c]: Well not exactly. I can use open(). I thought that using low level IO wasn't recommended in builtin/* but now I can see that there are traces. Thus will now use open(). Thanks!

File: builtin/bisect--helper.c (hunk 2 above)
Comment (pranitbauva1997) [inline builtin/bisect--helper.c]: Yes I agree that using strbuf is wasteful. I will remove this. I can just return "HEAD" or "BISECT_HEAD". Will update! Thanks!

File: builtin/bisect--helper.c (hunk 1 above)
Comment (chriscool) [inline builtin/bisect--helper.c]: Please take a look at the functions that are sometimes used along with with git_path_*().

Comment (pranitbauva1997) [inline builtin/bisect--helper.c]: Thanks! Will take a look.

**Pull request closed without merging**